content/terraform/v1.12.x/docs/language/meta-arguments.mdx

```diff
@@ -0,0 +1,211 @@
+---
+page_title: Meta-arguments
+description: Meta-arguments are built-in Terraform arguments that determine how Terraform creates and manages resources.
+---
+
+# Meta-arguments
+
+Meta-arguments are a class of arguments built into the Terraform configuration language that control how Terraform creates resources. This page provides an overview of the meta-arguments included with Terraform.
+
+## Introduction
+
+You can use meta-arguments in any type of `resource`  block, including resources created with `data` blocks. You can also use most meta-arguments in `module` blocks.
```

Comment:
You can also use meta-arguments in none-resource blocks (providers, checks, outputs, etc). Can you make this whole section more block agnostic? As in, make this more about how meta-arguments are built into the language and not defined by providers.